### ISSUE TITLE
All-day calendar events have wrong end-date in iCalendar feed

### ISSUE THREAD

**Issue by MikeN123** [instructure/canvas-lms#1201]
### Summary:

When an all-day calendar event is created, in the accompanying iCalendar feed the DTSTART and DTEND property have the same value. The DTEND value should be one day later than the DTSTART value.

### Steps to reproduce:

 1. Create an all-day event (event without start/end time) in the calendar
 2. Download the iCalender feed
 3. Compare the DTSTART and DTEND values

### Expected behavior:

DTEND should be one day after DTSTART value, for an all-day event with the duration of one day.

### Additional notes:

See RFC 5545 3.6.1:

>  The "DTEND" property for a "VEVENT" calendar component specifies the non-inclusive end of the event.

**Comment by spencerolson** [comment]
Hi @MikeN123 , thanks for bringing this to our attention. I've created a ticket for this and will put it in the appropriate team's backlog, where they can appropriately prioritize it among other tickets and determine the severity of the issue.

I did some manual testing with this, and it looks like Calendar App, Google Calendar, and Outlook (those are the only three I tested) handle the ICS file as-is without issue and interpret the event correctly as an all-day event, which is good. That being said, we should still make this change to adhere to RFC 5545 3.6.1 (thanks for providing the reference). 

Thanks,
Spencer
